app/src/main/java/de/westnordost/streetcomplete/data/download/tiles/DownloadedTilesDao.kt

```diff
@@ -31,16 +31,20 @@ class DownloadedTilesDao(private val db: Database) {
         db.exec("DELETE FROM $NAME")
     }
 
-    fun updateTime(tile: TilePos, time: Long) {
+    fun updateTimeIfNewer(tile: TilePos, time: Long) {
         db.update(NAME,
             values = listOf(DATE to time),
-            where = "$X = ? AND $Y = ?",
-            args = arrayOf(tile.x.toString(), tile.y.toString()),
+            where = "$X = ? AND $Y = ? AND $DATE > ?",
+            args = arrayOf(tile.x.toString(), tile.y.toString(), time.toString()),
         )
     }
 
-    fun updateAllTimes(time: Long) {
-        db.update(NAME, values = listOf(DATE to time))
+    fun updateAllTimesIfNewer(time: Long) {
```

Comment:
Sounds odd. Wouldn't `updateAllTimesNewerThan` be more descriptive? (Same for the other function, `updateTimeIfNewerThan` or something.)